pagopa/io-app
# chore: [IOBP-340] Add loading state to `BarcodeScanBaseScreenComponent`

## Short description
This PR implements the logic to enable the loading state to `BarcodeScanBaseScreenComponent`, which allows to display a loading spinner when the screen is currently occupied with some time intensive activity (decoding barcodes or presenting dialogs).

https://github.com/pagopa/io-app/assets/6160324/a062b4bd-4967-4f2e-97b7-89826a64c240

## List of changes proposed in this pull request
- [ts/features/barcode/components/AnimatedCameraMarker.tsx](https://github.com/pagopa/io-app/compare/IOBP-338-add-loading-state-to-barcode-scan-screen?expand=1#diff-9a7ca968e85405f3c0468404c3d7592a8f243f9ea801afb6ff5f2f6ee6073ddf): changed `state` prop to `isLoading`, used to enable loading state of the component 
- [ts/features/barcode/components/BarcodeScanBaseScreenComponent.tsx](https://github.com/pagopa/io-app/compare/IOBP-338-add-loading-state-to-barcode-scan-screen?expand=1#diff-1e4e400edd0a8debdd16a6c728d260d849b977083ea5aa108a5ba43ce25ac4c2): added `isLoading` and `isDisabled` props
- [useIOBarcodeCameraScanner.tsx](https://github.com/pagopa/io-app/compare/IOBP-338-add-loading-state-to-barcode-scan-screen?expand=1#diff-f9a213668da7a0880622ac4d0690f61c85abd9d961b6b1c05baafdf0717b5957): added `isLoading` and `isDisabled` props, added an animated `LoadingSpinner` if `isLoading` is set to true 
- [ts/features/barcode/hooks/useIOBarcodeFileReader.tsx](https://github.com/pagopa/io-app/compare/IOBP-338-add-loading-state-to-barcode-scan-screen?expand=1#diff-36bd07843535011c4a091fd48fba32d133ba39986aa714dbe03a54332c1803a1): added additional information in the returned object to indicate whether the hook is currently occupied with decoding (`isLoading`) or is presenting some picker (`isFilePickerVisible`)
- [ts/features/barcode/screens/BarcodeScanScreen.tsx](https://github.com/pagopa/io-app/compare/IOBP-338-add-loading-state-to-barcode-scan-screen?expand=1#diff-12570b34029e534a64b24f1cb81383d99557273c4b2f34ad4ced82685fa017f6): alignment with new hooks and components props 
- [ts/features/idpay/payment/screens/IDPayPaymentCodeScanScreen.tsx](https://github.com/pagopa/io-app/compare/IOBP-338-add-loading-state-to-barcode-scan-screen?expand=1#diff-eaf783c2cc9816aca220690082f4ba077df5954553a99078eb5e3faec9bd4b27): alignment with new hooks and components props 
- [ts/features/walletV3/payment/screens/WalletPaymentBarcodeScanScreen.tsx](https://github.com/pagopa/io-app/compare/IOBP-338-add-loading-state-to-barcode-scan-screen?expand=1#diff-627f530afb8e81f8849fbfbfa2ee5ece0434bdbe062355178c9df587151a9513): alignment with new hooks and components props 
- [ts/features/walletV3/payment/screens/WalletPaymentBarcodeChoiceScreen.tsx](https://github.com/pagopa/io-app/pull/5166/files#diff-df9889d94972386628b359910ce7aa1909a30d8a8d95b7931cf68fdc46499483): removed unnecessary nested `FlatList`

## How to test
Navigate to any barcode scan screen (from home or from ID Pay initiative details screen) and check the following:
- the marker animation is disabled whenever a picker dialog is showed
- the loading spinner is visible when a barcode is being decoded (from an image or a file) or a picker is waiting to be displayed
